qimgv: cudaSupport

### Describe the bug
qimgv fails to build when `nixpkgs.config.cudaSupport = true`.

```console
CMake Error at /nix/store/m33sn6h5kgfbk3z4avwmgxam1dny07i0-cmake-3.21.2/share/cmake-3.21/Modules/FindCUDA.cmake:859 (message):
  Specify CUDA_TOOLKIT_ROOT_DIR
Call Stack (most recent call first):
  /nix/store/jrqws14nfflrja5bgdq64fizmnphirhz-opencv-4.5.2/lib/cmake/opencv4/OpenCVConfig.cmake:86 (find_package)
  /nix/store/jrqws14nfflrja5bgdq64fizmnphirhz-opencv-4.5.2/lib/cmake/opencv4/OpenCVConfig.cmake:108 (find_host_package)
  CMakeLists.txt:66 (find_package)


-- Configuring incomplete, errors occurred!
See also "/build/source/build/CMakeFiles/CMakeOutput.log".
builder for '/nix/store/p8b87g5kxgy0i3p5piry1zldyw27d2nf-qimgv-1.0.1.drv' failed with exit code 1
error: build of '/nix/store/p8b87g5kxgy0i3p5piry1zldyw27d2nf-qimgv-1.0.1.drv' failed
```

### Steps To Reproduce
Steps to reproduce the behavior:
```nix
{
  pkgs ? import <nixpkgs> {
    config = {
      allowUnfree = true;
      cudaSupport = true;
    };
  }
}:

pkgs.stdenv.mkDerivation {
  name = "qimgv-cuda";
  buildInputs = with pkgs; [
    qimgv
  ];
}
```

### Expected behavior
No build errors.

### Screenshots
N/A.

### Additional context
N/A.

### Notify maintainers
@cole-h

### Metadata
```console
λ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.73, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11pre323790.5f0194220f2"`
 - channels(mulhaq): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: qimgv
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/libraries/opencv/3.x.nix
+++ b/pkgs/development/libraries/opencv/3.x.nix
@@ -213,11 +213,11 @@ stdenv.mkDerivation {
     # tesseract & leptonica.
     ++ lib.optionals enableTesseract [ tesseract leptonica ]
     ++ lib.optional enableTbb tbb
-    ++ lib.optional enableCuda cudatoolkit
     ++ lib.optionals stdenv.isDarwin [ bzip2 AVFoundation Cocoa VideoDecodeAcceleration ]
     ++ lib.optionals enableDocs [ doxygen graphviz-nox ];
 
-  propagatedBuildInputs = lib.optional enablePython pythonPackages.numpy;
+  propagatedBuildInputs = lib.optional enablePython pythonPackages.numpy
+    ++ lib.optional enableCuda cudatoolkit;
 
   nativeBuildInputs = [ cmake pkg-config unzip ];
 

--- a/pkgs/development/libraries/opencv/4.x.nix
+++ b/pkgs/development/libraries/opencv/4.x.nix
@@ -269,11 +269,11 @@ stdenv.mkDerivation {
     # tesseract & leptonica.
     ++ lib.optionals enableTesseract [ tesseract leptonica ]
     ++ lib.optional enableTbb tbb
-    ++ lib.optionals enableCuda [ cudatoolkit nvidia-optical-flow-sdk ]
     ++ lib.optionals stdenv.isDarwin [ bzip2 AVFoundation Cocoa VideoDecodeAcceleration CoreMedia MediaToolbox ]
     ++ lib.optionals enableDocs [ doxygen graphviz-nox ];
 
-  propagatedBuildInputs = lib.optional enablePython pythonPackages.numpy;
+  propagatedBuildInputs = lib.optional enablePython pythonPackages.numpy
+    ++ lib.optionals enableCuda [ cudatoolkit nvidia-optical-flow-sdk ];
 
   nativeBuildInputs = [ cmake pkg-config unzip ];
 

